### PR TITLE
Updated Program APIs for WP

### PR DESCRIPTION
**Pull request by manan-memon**
This PR updates the Program APIs for to be used on WordPress.
JIRA: https://edlyio.atlassian.net/browse/EDLY-7129

### CODE REVIEW

File: course_discovery/apps/api/serializers.py [hunk 1]
```diff
@@ -813,7 +813,7 @@ class Meta:
         model = CourseRun
         fields = ('key', 'uuid', 'title', 'external_key', 'image', 'short_description', 'marketing_url',
                   'seats', 'start', 'end', 'go_live_date', 'enrollment_start', 'enrollment_end',
-                  'pacing_type', 'type', 'run_type', 'status', 'is_enrollable', 'is_marketable', 'term', 'subjects',)
+                  'pacing_type', 'type', 'run_type', 'status', 'is_enrollable', 'is_marketable', 'term', 'subjects', 'card_image_url')
```
Comment (taimoor-ahmed-1) [inline course_discovery/apps/api/serializers.py]: nit: line should shouldn't be greater than 120 chars.

File: course_discovery/apps/course_metadata/admin.py [hunk 2]
```diff
@@ -203,7 +203,7 @@ class ProgramAdmin(admin.ModelAdmin):
         'card_image', 'marketing_slug', 'overview', 'credit_redemption_overview', 'video', 'total_hours_of_effort',
         'weeks_to_complete', 'min_hours_effort_per_week', 'max_hours_effort_per_week', 'courses',
         'order_courses_by_start_date', 'custom_course_runs_display', 'excluded_course_runs', 'authoring_organizations',
-        'credit_backing_organizations', 'one_click_purchase_enabled', 'hidden', 'corporate_endorsements', 'faq',
+        'credit_backing_organizations', 'one_click_purchase_enabled', 'hidden', 'featured', 'corporate_endorsements', 'faq',
```
Comment (taimoor-ahmed-1) [inline course_discovery/apps/course_metadata/admin.py]: nit: line seems greater 120 characters

**Review by taimoor-ahmed-1 — CHANGES_REQUESTED** [review]

File: course_discovery/apps/course_metadata/search_indexes.py [hunk 3]
```diff
@@ -410,6 +418,38 @@ def prepare_search_card_display(self, obj):
...
+        if course_runs:
+            first_course_run = course_runs[0]
+            first_seat = first_course_run.seats.first()
+            if first_seat:
+                return first_seat.currency.code
+        return None
+    
+    def prepare_title_override(self, obj):
+        return obj.title.title()
+
```
Comment (Anas-hameed) [inline course_discovery/apps/course_metadata/search_indexes.py]: nit: there should be two blank lines at the class definition end. Remove the extra empty line.

**Review by Anas-hameed — CHANGES_REQUESTED** [review]

File: course_discovery/apps/api/serializers.py (hunk 1 above)
Comment (manan-memon) [reply course_discovery/apps/api/serializers.py]: Done

File: course_discovery/apps/course_metadata/admin.py (hunk 2 above)
Comment (manan-memon) [reply course_discovery/apps/course_metadata/admin.py]: Changed

File: course_discovery/apps/course_metadata/search_indexes.py (hunk 3 above)
Comment (manan-memon) [reply course_discovery/apps/course_metadata/search_indexes.py]: Removed

File: course_discovery/apps/course_metadata/search_indexes.py [hunk 4]
```diff
@@ -410,6 +418,37 @@ def prepare_search_card_display(self, obj):
...
         return [degree.search_card_ranking, degree.search_card_cost, degree.search_card_courses]
+    
+    def prepare_num_of_courses(self, obj):
+        return obj.courses.count()
+    
+    def prepare_banner_image_url(self, obj):
+        if obj.banner_image_url:
+            return obj.banner_image_url
+        elif obj.banner_image:
+            return obj.banner_image.url
```
Comment (Anas-hameed) [inline course_discovery/apps/course_metadata/search_indexes.py:429]: what if we don't have `obj.banner_image.url`, Will the code break?

File: course_discovery/apps/course_metadata/search_indexes.py [hunk 5]
```diff
@@ -410,6 +418,37 @@ def prepare_search_card_display(self, obj):
...
+        return None
+    
+    def prepare_bundle_price(self, obj):
+        bundle_price = 0
+        course_runs = list(obj.course_runs)
+        for course_run in course_runs:
+            first_enrollable_paid_seat_price = course_run.first_enrollable_paid_seat_price
+            if first_enrollable_paid_seat_price:
+                bundle_price += first_enrollable_paid_seat_price
+        return bundle_price
```
Comment (Anas-hameed) [inline course_discovery/apps/course_metadata/search_indexes.py]: This loop can be simplified.

File: course_discovery/apps/course_metadata/search_indexes.py [hunk 6]
```diff
@@ -410,6 +418,37 @@ def prepare_search_card_display(self, obj):
...
+    
+    def prepare_bundle_currency(self, obj):
+        course_runs = list(obj.course_runs)
+        if course_runs:
+            first_course_run = course_runs[0]
+            first_seat = first_course_run.seats.first()
+            if first_seat:
+                return first_seat.currency.code
+        return None
```
Comment (Anas-hameed) [inline course_discovery/apps/course_metadata/search_indexes.py]: Can we do this in reverse order. Unhappy path first.  e.g None should be returned first.

**Review by Anas-hameed — CHANGES_REQUESTED** [review]

File: course_discovery/apps/course_metadata/search_indexes.py (hunk 4 above)
Comment (manan-memon) [reply course_discovery/apps/course_metadata/search_indexes.py:429]: The function will return None in that case (not explicitly mentioned), this won't break anything as the field is nullable.

File: course_discovery/apps/course_metadata/search_indexes.py (hunk 5 above)
Comment (manan-memon) [reply course_discovery/apps/course_metadata/search_indexes.py]: Simplified as discussed

File: course_discovery/apps/course_metadata/search_indexes.py (hunk 6 above)
Comment (manan-memon) [reply course_discovery/apps/course_metadata/search_indexes.py]: Changed

**Review by taimoor-ahmed-1 — APPROVED** [review]

**Review by Anas-hameed — APPROVED** [review]
LGTM

**Pull request merged**